### ISSUE TITLE
Parse error with CASE

### ISSUE THREAD

**Issue by stoch** [K2InformaticsGmbH/sqlparse#56]
@c-bik Is this simply not supported?  If not, could it be?
```sql
select MAX( CASE WHEN ROWNUM = 10 THEN 11 ELSE 0 END ) as RESULT FROM WARNING WHERE ROWNUM <= 10
```
```json
{
        "logTime": "19.03.2017 11:50:37.946675",
        "logLevel": "error",
        "pid": "<0.9658.31>",
        "module": "gen_adapter",
        "function": "process_cmd",
        "line": 148,
        "message": "parse_stmt error in parsetree {\"3: syntax error before: 'CASE'\",[{'SELECT',1},{'FUNS',2,'MAX'},{'(',2},{'CASE',3},{'WHEN',4},{'NAME',6,\"ROWNUM\"},{'COMPARISON',4,'='},{'INTNUM',4,\"10\"},{'THEN',4},{'INTNUM',4,\"11\"},{'ELSE',5},{'INTNUM',6,\"0\"},{'END',7},{')',8},{'AS',8},{'NAME',6,\"RESULT\"},{'FROM',9},{'NAME',10,\"ALL_TABLES\"},{'WHERE',10},{'NAME',6,\"ROWNUM\"},{'COMPARISON',10,'<='},{'INTNUM',10,\"20\"},{';',10}]}",
        "stacktrace": "[]"
}
```
Also happens for simplified SQL.
```sql
select CASE WHEN ROWNUM = 10 THEN 11 ELSE 0 END  FROM WARNING WHERE ROWNUM <= 10
```
CASE is preferable to DECODE. All new MTRAC queries use it.

**Comment by c-bik** [comment]
@stoch Grammar rules for SQL `CASE` are incomplete/missing. I will work on it for the next version!

**Comment by walter-weinmann** [comment]
Solved by #90.
